File: cinema-backend/src/main/java/org/example/cinemabackend/cinema/application/dto/response/CinemaResponse.java
package org.example.cinemabackend.cinema.application.dto.response;

import lombok.Builder;
import org.example.cinemabackend.user.application.dto.response.UserResponse;

import java.util.Set;

@Builder
public record CinemaResponse(
        String name,
        String description,
        AddressResponse address,
        ImageResponse image,
        Set<ContactDetailsResponse> contactDetails,
        UserResponse cinemaManager
) {
}

File: cinema-frontend/src/app/cinema/dtos/response/cinema.response.ts
import {AddressResponse} from "./address.response";
import {ImageResponse} from "../../../movie/dtos/response/image.response";
import {ContactDetailsResponse} from "./contact-details.response";
import {ScreeningRoomResponse} from "../../../repertory/dtos/screening-room.response";
import {UserResponse} from "../../../user/dtos/response/user.response";

export class CinemaResponse {
  constructor(
    readonly name: string,
    readonly description: string,
    readonly address: AddressResponse,
    readonly image: ImageResponse,
    readonly screeningRooms: ScreeningRoomResponse[],
    readonly contactDetails: ContactDetailsResponse[],
    readonly cinemaManager: UserResponse,
  ) {
  }
}
